drivers/i2c/i2c_mcux_flexcomm.c

```diff
@@ -35,6 +35,7 @@ struct mcux_flexcomm_config {
 struct mcux_flexcomm_data {
 	i2c_master_handle_t handle;
 	struct k_sem device_sync_sem;
+	struct k_sem lock;
```

Comment:
why use a semaphore here versus a mutex? I suppose here that a mutex isn't required, since the only consumer is the driver

Reply:
Waiting to take a binary semaphore doesn't make the thread eligible for priority inheritance from what I understand compared to a mutex, and the documentation about mutexes suggests that it is bad practice to have a thread lock multiple mutexes because "sub-optimal behaviour will be observed". So therefore from this understanding that I have from the kernel documentation, I think the use of mutexes should be left to the application and binary semaphores are more appropriate for driver development.

https://docs.zephyrproject.org/latest/kernel/services/synchronization/mutexes.html#api-reference
https://docs.zephyrproject.org/latest/kernel/services/synchronization/semaphores.html#api-reference

Reply:
If its any indicator for consistency, SPI uses a semaphore in spi_context for all spi drivers to serialize access.

Reply:
You make a good point about priority inheritance, especially with multiple mutexes. Given that, semaphores do seem like the right call here.